arangod/Graph/BreadthFirstEnumerator.cpp

```diff
@@ -297,19 +302,33 @@ bool BreadthFirstEnumerator::prepareSearchOnNextDepth() {
 
 bool BreadthFirstEnumerator::shouldPrune() {
   if (_opts->usesPrune()) {
-    auto* evaluator = _opts->getPruneEvaluator();
-    if (evaluator->needsVertex()) {
-      evaluator->injectVertex(vertexToAqlValue(_schreierIndex).slice());
-    }
-    if (evaluator->needsEdge()) {
-      evaluator->injectEdge(edgeToAqlValue(_schreierIndex).slice());
-    }
-    transaction::BuilderLeaser builder(_opts->trx());
-    if (evaluator->needsPath()) {
-      aql::AqlValue val = pathToIndexToAqlValue(*builder.get(), _schreierIndex);
-      evaluator->injectPath(val.slice());
+    // evaluator->evaluate() might access these, so they have to live long enough.
+    // To make that perfectly clear, I added a scope.
+    transaction::BuilderLeaser pathBuilder(_opts->trx());
+    aql::AqlValue vertex, edge;
+    aql::AqlValueGuard vertexGuard{vertex, true}, edgeGuard{edge, true};
+    {
+      auto *evaluator = _opts->getPruneEvaluator();
```

Comment:
It can't, `usesPrune()` assures that. The dtor has the nullptr check only because I used it first there, and added the `usesPrune()` check only later, without removing the now superfluous nullptr check.